src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2270,6 +2301,36 @@ private void ReceiveMessageFromAsync_Helper(IPAddress listenOn, IPAddress connec
 
         #endregion ReceiveMessageFrom
 
+        [Fact]
+        [PlatformSpecific(PlatformID.OSX)]
+        public void BeginReceiveFrom_NotSupported()
+        {
+            using (Socket sock = new Socket(SocketType.Dgram, ProtocolType.Udp))
+            {
+                EndPoint ep = new IPEndPoint(IPAddress.Any, 0);
+                sock.Bind(ep);
+
+                byte[] buf = new byte[1];
+
+                Assert.Throws<PlatformNotSupportedException>(() => sock.BeginReceiveFrom(buf, 0, buf.Length, SocketFlags.None, ref ep, null, null));
```

Comment:
Yes, we do have some coverage of these methods, though we could probably use more testing of the various possibilities here.  I've created #6676 to track this.